source/os_specific/service_layers/oszephyr.c

```diff
@@ -967,7 +967,10 @@ UINT64
 AcpiOsGetTimer (
     void)
 {
-    return (k_cycle_get_64 ());
+    uint32_t TimerHi, TimerLo;
+
+    __asm__ volatile("rdtsc" : "=d"(TimerHi), "=a"(TimerLo));
```

Comment:
again you are making this x86 specific. 

Let's keep k_cycle_get_64().
And if we have such race condition, it means we are not initializing hpet (or any other timer actually) early enough. So it's a zephyr side issue, not an acpica one.